### ISSUE TITLE
Add Errors()

### ISSUE THREAD

**Issue by elliotchance** [elliotchance/tf#1]
Based on the fact many functions return `result, error` it can test for the existence of an error or even further with a message:

```go
Divide(5, 0).Errors() // pass
Divide(5, 0).Errors("divide-by-zero") // pass
Divide(5, 0).Errors("foo") // fail
```

**Comment by m1ome** [comment]
In Golang errors are values, more comfortable to provide error with a particular message and type, and check them

**Comment by elliotchance** [comment]
What do you mean by error type?

**Comment by m1ome** [comment]
```go

type MyError struct {
    Err error
}

func (e MyError) Error() string {
    return err.Error()
}

customErr := MyError{
    Err: errors.New("i am an error wrapper")
}

r := tf.Function(t, ReturnError() error {
    return customError
})

// This wil pass
r().Errors("i am an error wrapper")
r().Errors(customError)
r().Errors()

// This will fail
r().Errors(errors.New("i am an error wrapper"))
```

**Comment by elliotchance** [comment]
Ah yes, I understand what you mean now. I agree.

**Comment by m1ome** [comment]
@elliotchance this one solved by #12

**Comment by elliotchance** [comment]
Thanks Pavel.
